source/geometries/Next100Vessel.cc

```diff
@@ -366,21 +366,23 @@ namespace nexus {
     }
 
     // VERTEX GENERATORS   //////////
-    body_gen_  = new CylinderPointSampler(vessel_in_rad_, body_length_, vessel_thickness_, 0.);
+    body_gen_  = new CylinderPointSampler2020(vessel_in_rad_, vessel_out_rad, body_length_/2.,
+                                              0., 360.*deg, 0, G4ThreeVector(0., 0., 0.));
 
     energy_endcap_gen_ = new SpherePointSampler( endcap_in_rad_, vessel_thickness_, energy_endcap_pos, 0,
 						   0., twopi, 0., endcap_theta_);
 
     tracking_endcap_gen_ = new SpherePointSampler( endcap_in_rad_, vessel_thickness_, tracking_endcap_pos, xRot,
 						 0., twopi, 0., endcap_theta_);
 
-    tracking_flange_gen_  = new CylinderPointSampler(vessel_out_rad, flange_tp_length,
-						     flange_out_rad-vessel_out_rad, 0., tracking_flange_pos);
+    tracking_flange_gen_ = new CylinderPointSampler2020(vessel_in_rad_, flange_out_rad, flange_tp_length/2.,
```

Comment:
Is it possible to use the constructor that takes the physical volume? The idea of this new class of generators is to use the physical volume, whenever is possible, to avoid potential mistakes.

Reply:
Probably, but:

1) the generator must go in Next100.cc since is where the physical volume is defined; and
2) since it is a cilindrical generator I understand that it would create a cilinder covering the whole vessel volume and just keep the vertices inside the vessel. I remember a conversation in which we should avoid that because it can be inneficient due to most of the randoms are discarded.

That's why I maintained the current approach, should I change it anyway?

Reply:
I see that the physical volume is the union of the several volumes that make up the vessel; in that case it's better to leave it as you implemented it, because that volume is not a cylinder. I don't understand your point 2: the generator reads the minimum and maximum radius, so if the cylinder is hollow, it takes that into account.

Reply:
Right, the point is that the volume is not a cylinder: it is an union of a cylinder, the flanges, the endcaps and the port nozzles. Then, do we agree to left it as it is?

Reply:
Yes.